### PR TITLE
[CELEBORN-1725][FOLLOWUP] Optimize `isAllMapTasksEnd` performance

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

Followup for https://github.com/apache/celeborn/pull/2905,

using the same logic to optimize `isAllMapTasksEnd` method.

### Why are the changes needed?
Address comments: https://github.com/apache/celeborn/pull/2905#pullrequestreview-2457905880

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Same logic with https://github.com/apache/celeborn/pull/2905

### CODE REVIEW

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala [hunk 1]
```diff
@@ -862,8 +862,16 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       return
     }
 
-    def isAllMaptaskEnd(shuffleId: Int): Boolean = {
-      !commitManager.getMapperAttempts(shuffleId).exists(_ < 0)
+    def isAllMapTasksEnd(shuffleId: Int): Boolean = {
```
Comment (RexXiong) [inline client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:865]: Seems we can eliminate these duplicate code.

**Review by onebox-li — COMMENTED** [review]
Thanks! I think this code fragment could be polished too . https://github.com/apache/celeborn/blob/main/worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala#L455

**Comment by FMX** [comment]
You can add a new PR to fix all similar code snippets. There are still some more `exists` codes. 
For example, there are similar codes in `ReducePartitionCommitHandler`.

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala (hunk 1 above)
Comment (turboFei) [reply client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala]: done

**Comment by turboFei** [comment]
> You can add a new PR to fix all similar code snippets. There are still some more `exists` codes. For example, there are similar codes in `ReducePartitionCommitHandler`.

Ok, thx

**Comment by turboFei** [comment]
> Thanks! I think this code fragment could be polished too . https://github.com/apache/celeborn/blob/main/worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala#L455-L466

Thanks for the suggestion, will address it in CELEBORN-1753

File: client/src/main/scala/org/apache/celeborn/client/ClientUtils.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.client
+
+object ClientUtils {
+  def areAllMapperAttemptsFinished(attempts: Array[Int]): Boolean = {
```
Comment (pan3793) [inline client/src/main/scala/org/apache/celeborn/client/ClientUtils.scala:21]: I think we need some comments to explain the parameter and why this is fast

Comment (turboFei) [reply client/src/main/scala/org/apache/celeborn/client/ClientUtils.scala:30]: thanks, addressed

File: client/src/main/scala/org/apache/celeborn/client/ClientUtils.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+package org.apache.celeborn.client
+
+object ClientUtils {
+
+  /**
+   * Check if all the mapper attempts are finished. If any of the attempts is not finished, return false.
+   * This method checks the attempts array in reverse order, which can be faster if the unfinished attempts
+   * are more likely to be towards the end of the array.
+   *
+   * @param attempts The mapper attempts.
```
Comment (pan3793) [inline client/src/main/scala/org/apache/celeborn/client/ClientUtils.scala:27]: could you please also describe what the value means?

Comment (turboFei) [reply client/src/main/scala/org/apache/celeborn/client/ClientUtils.scala]: done

**Review by pan3793 — APPROVED** [review]

**Review by onebox-li — APPROVED** [review]

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
Thanks, merge to main(V0.6.0) and branch-0.5(v0.5.3) and branch-0.4(v0.4.3)

**Pull request closed without merging**